Sources/SmithyXML/Reader/Reader+libxml2.swift

```diff
@@ -0,0 +1,157 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import libxml2
+import struct Foundation.Data
+
+extension Reader {
```

Comment:
These extensions use libxml2 to parse XML into a tree of `Reader` objects that can be used to deserialize a HTTP response body.